python/lsst/ts/ess/ess_csc.py

```diff
@@ -36,21 +59,90 @@ def __init__(
             simulation_mode=simulation_mode,
         )
 
-        stream_handler = logging.StreamHandler()
-        self.log.addHandler(stream_handler)
         self.log.info("__init__")
 
+        self._ess_sensor = None
+        self.temperature_task = None
+
+    async def get_temperature(self, interval):
+        """Get the temperature forever at the specified interval.
+
+        Parameters
+        ----------
+        interval: `float`
+            The interval (sec) at which to get the temperature.
+
+        Returns
+        -------
+
+        """
+        try:
+            previous_tai = 0
+            while True:
+                self.log.debug("Getting the temperature from the sensor")
+                await self._ess_sensor.readInstrument()
+                data = self._ess_sensor.myData
+                if data.timestamp - previous_tai > 1.0:
+                    previous_tai = data.timestamp
+                    telemetry = {}
+                    telemetry["timestamp"] = data.timestamp
+                    telemetry["TemperatureC01"] = data.TemperatureC01
```

Comment:
Switching TemeratureC to array will allow for a for loop setting those value (

```python
for in in range(1,5):
    telemetry[f"TemperatureC{i:02}"] = data.temperatureC[i]

Reply:
I agree that that would be better but I don't know what the internal structure of this class is so until I know that I cannot change this code. I am merely following what's done in the ESS_Temperature4Ch_Publisher.sh file. And I will work with our external developer to improve the code in this respect, including camel casing of the variable names.